### ISSUE TITLE
Decoder generator failure []struct

### ISSUE THREAD

**Issue by klauspost** [pquerna/ffjson#40]
The following file fails to generate valid code:

```
package test

type Custom struct {
	Subdivisions []struct {
		Name string
	}
}
```

The failing part of the generated code looks like this:

```
	if tok == fflib.FFTok_null {
		uj.Subdivisions = nil
	} else {	
-->>		uj.Subdivisions = make([], 0)
	}
```

**Comment by moorereason** [comment]
I'm seeing the same issue with a slice of slices or a slice of maps.  The following code:

```go
type TestSliceSlice struct {
        Records [][]string
}
type TestSliceMap struct {
        Records []map[string]string
}
```

Generates invalid code:

```go
        if tok == fflib.FFTok_null {
                uj.Records = nil
        } else {

                uj.Records = make([], 0)

        }
```

**Comment by pquerna** [comment]
kk.  I think i'll try to have it just use the fallback path for some of these.

**Comment by klauspost** [comment]
Fixed by PR #92
